pkg/controller/srcdst_controller.go

```diff
@@ -58,66 +64,121 @@ func NewSrcDstController(client kubernetes.Interface, ec2Client *ec2.EC2) *Contr
 	srcDstCheckEnabled := true
 	if node.Annotations != nil {
 		if _, ok := node.Annotations[SrcDstCheckDisabledAnnotation]; ok {
 			srcDstCheckEnabled = false
 		}
 	}
 
+	glog.V(6).Infof("Performing disableSrcDstIfEnabled run for node %s", node.Name)
+
 	if srcDstCheckEnabled {
+
+		glog.V(5).Infof("srcDstCheckEnabled true for node %s", node.Name)
+
 		// src dst check disabled annotation does not exist
 		// call AWS ec2 api to disable
-		instanceID, err := GetInstanceIDFromProviderID(node.Spec.ProviderID)
-		if err != nil {
-			glog.Errorf("Fail to retrieve Instance ID from Provider ID: %v", node.Spec.ProviderID)
-			return
-		}
-		err = c.disableSrcDstCheck(*instanceID)
+		glog.V(5).Infof("Calling AWS to disable Node SrcDstCheck for node %s", node.Name)
+		err, instanceID := c.disableSrcDstCheck(node)
 		if err != nil {
 			glog.Errorf("Fail to disable src dst check for EC2 instance: %v; %v", *instanceID, err)
 			return
 		}
+
 		// We should not modify the cache object directly, so we make a copy first
 		nodeCopy, err := common.CopyObjToNode(node)
 		if err != nil {
 			glog.Errorf("Failed to make copy of node: %v", err)
 			return
 		}
+
 		glog.Infof("Marking node %s with SrcDstCheckDisabledAnnotation", node.Name)
 		nodeCopy.Annotations[SrcDstCheckDisabledAnnotation] = "true"
-		if _, err := c.client.Core().Nodes().Update(nodeCopy); err != nil {
-			glog.Errorf("Failed to set %s annotation: %v", SrcDstCheckDisabledAnnotation, err)
+		if my_opts.Patchnode {
+			glog.V(5).Infof("Patching node %s", nodeCopy.Name)
+			// Thanks to https://github.com/tamalsaha/patch-demo/blob/master/main.go#L113 for this stanza
+			// Prep JSON for the newly changed node object
+			json_nodeCopy, err := json.Marshal(nodeCopy)
+			if err != nil {
+				glog.Errorf("Failed to marshal nodeCopy into JSON, %v", err)
+			}
+			// Prep JSON for a copy of the old node object
+			nodeCopyOrig, err := common.CopyObjToNode(node)
+			if err != nil {
+				glog.Errorf("Failed to make an original copy of node: %v", err)
+				return
+			}
+			json_nodeCopyOrig, err := json.Marshal(nodeCopyOrig)
+			if err != nil {
+				glog.Errorf("Failed to marshal nodeCopyOrig into JSON, %v", err)
+			}
+			// Prepare the JSON patch
+			patch, err := jsonpatch.CreatePatch(json_nodeCopyOrig, json_nodeCopy)
+			if err != nil {
+				glog.Errorf("Failed to CreatePatch between JSONs, %v", err)
+			}
+			// Fix indenting
+			json_patch, err := json.MarshalIndent(patch, "", "  ")
+			if err != nil {
+				glog.Errorf("Failed to MarshalIndent: %v", err)
+			}
+			glog.V(5).Info(string(json_patch))
+			// https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/node.go#L170
+			if _, err := c.client.Core().Nodes().Patch(nodeCopy.Name, types.JSONPatchType, json_patch); err != nil {
+				glog.Errorf("Failed to patch %s annotation: %v", SrcDstCheckDisabledAnnotation, err)
+				// Sleep for a second and give it one
```

Comment:
This comment is slightly oddly formatted with so many line breaks. It also doesn't make clear to me in what circumstance the retry after 1 second would handle an initial failure. The reference to AWS is also confusing me here as the calls to AWS' APIs occur outside of this logic loop.